lib/mongoid/extensions/typed_array.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+module Mongoid
+  class TypedArray < ::Array
```

Comment:
This isn't an extension, TypedArray a type defined by Mongoid. It should be in `lib/mongoid/typed_array.rb`.